### ISSUE TITLE
Documentation build fails in v2.8.1

### ISSUE THREAD

**Issue by orlitzky** [wbhart/flint2#1005]
A Gentoo user noticed this problem in [bug 816822](https://bugs.gentoo.org/816822) and I am able to reproduce it. Building the docs with CMake, we see errors like,

```
Running Sphinx v4.1.2
making output directory... done
building [mo]: targets for 0 po files that are out of date
building [html]: targets for 91 source files that are out of date
updating environment: [new config] 91 added, 0 changed, 0 removed
...
/var/tmp/portage/sci-mathematics/flint-2.8.1/work/flint-2.8.1/doc/source/flintxx_functions.rst:24: WARNING: Error in declarator or parameters
Invalid C declaration: Expected identifier in nested name. [error at 7]
  frandxx::frandxx()
  -------^
```

Similar errors repeat for some time in `flintxx_functions.rst`. At the end, a few more isolated errors are reported:

```
/var/tmp/portage/sci-mathematics/flint-2.8.1/work/flint-2.8.1/doc/source/fmpz_mat.rst:965: WARNING: Error in declarator
If declarator-id with parameters (e.g., 'void f(int arg)'):
  Invalid C declaration: Expected identifier in nested name. [error at 23]
    fmpz_mat_can_solve_fflu(fmpz_mat_t X, fmpz_t den, const fmpz_mat_t A, const fmpz_mat_t B)
    -----------------------^
If parenthesis in noptr-declarator (e.g., 'void (*f(int arg))(double)'):
  Error in declarator or parameters
  Invalid C declaration: Expecting "(" in parameters. [error at 35]
    fmpz_mat_can_solve_fflu(fmpz_mat_t X, fmpz_t den, const fmpz_mat_t A, const fmpz_mat_t B)
```

```
/var/tmp/portage/sci-mathematics/flint-2.8.1/work/flint-2.8.1/doc/source/fq_nmod.rst:255: WARNING: Error in declarator or parameters
Invalid C declaration: Expected identifier in nested name. [error at 5]
  Sets ``rop`` to the square root of ``op1`` if it is a square, and return
  -----^
/var/tmp/portage/sci-mathematics/flint-2.8.1/work/flint-2.8.1/doc/source/fq_nmod.rst:255: WARNING: Invalid C declaration: Expected identifier in nested name. [error at 0]
  `1`, otherwise return `0`.
  ^
```

```
/var/tmp/portage/sci-mathematics/flint-2.8.1/work/flint-2.8.1/doc/source/padic_mat.rst:57: WARNING: Duplicate C declaration, also defined at padic_mat:39.
Declaration is '.. c:function:: slong padic_mat_val(const padic_mat_t A)'.
```

**Comment by wbhart** [comment]
Did you take the tarball from the website? Apparently there were problems with it.

**Comment by wbhart** [comment]
Fixed in trunk.

**Comment by orlitzky** [comment]
Thanks!
